.github/workflows/matrix.yml

```diff
@@ -0,0 +1,62 @@
+name: Test Matrix
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [16, 18, 20]
+        operating-system:
+          - ubuntu-latest
+          - windows-latest
+          # - macos-latest -> Rollup is not supported on ARM64 - https://github.com/npm/cli/issues/4828
```

Comment:
There's a bug with Rollup and MacOS-latest. Basically it requires removal of the package-lock + node_modules and a clean install. Because of this, we're just going to skip running the matrix on MacOS-latest until its fixed.